src/pybind/mgr/telemetry/module.py

```diff
@@ -771,13 +775,15 @@ def gather_device_report(self) -> Dict[str, Dict[str, Dict[str, str]]]:
                 # this is a map of stamp -> {device info}
                 m = self.remote('devicehealth', 'get_recent_device_metrics',
                                 devid, min_sample)
-            except Exception:
+            except Exception as e:
+                self.log.debug('Unable to get recent metrics from a device: {}'.format(e))
                 continue
 
             # anonymize host id
             try:
                 host = d['location'][0]['host']
-            except (KeyError, IndexError):
+            except (KeyError, IndexError) as e:
+                self.log.debug("Unable to get host from a device: {}".format(e))
```

Comment:
let's add the device id to the message here too

Reply:
Fixed